tests/system_health/device_mocker.py

```diff
@@ -0,0 +1,62 @@
+import os
+import pytest
+import sys
+
+
+class DeviceMocker:
+    def deinit(self):
+        pass
+
+    def mock_fan_presence(self, status):
+        return False, None
+
+    def mock_fan_status(self, status):
+        return False, None
+
+    def mock_fan_speed(self, good):
+        return False, None
+
+    def mock_asic_temperature(self, good):
+        return False
+
+    def mock_psu_presence(self, status):
+        return False, None
+
+    def mock_psu_status(self, status):
+        return False, None
+
+    def mock_psu_temperature(self, good):
+        return False, None
+
+    def mock_psu_voltage(self, good):
+        return False, None
+
+
+@pytest.fixture
+def device_mocker_factory():
+    """
+    Fixture for system health data mocker factory.
+    :return: A function for creating system health related data mocker.
+    """
+    mockers = []
+
+    def _create_mocker(dut):
+        """
+        Create vendor specified mocker object by mocker name.
+        :param dut: DUT object representing a SONiC switch under test.
+        :return: Created mocker instance.
+        """
+        platform = dut.facts['platform']
+        mocker_object = None
+        if 'mlnx' in platform:
```

Comment:
It would be better to use `dut.facts['asic_type']` here.

Reply:
Changed.